### PR TITLE
ADCM-2150 Add tests for cluster and provider upgrade with group configs

### PR DESCRIPTION
**Pull request by dgusakov**

### CODE REVIEW

File: tests/functional/test_config_after_upgrade.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+class TestUpgradeWithConfigs:
+    """Tests for cluster and provider upgrades with ordinary configs"""
+
+    @pytest.mark.parametrize(
+        ("bundle_name", "update_func"),
+        [
+            ("cluster_for_upgrade_with_configs_with_new_fields", _update_config_and_attr_for_new_field_test),
+            (
+                "cluster_for_upgrade_with_configs_with_new_default_value",
+                _update_config_and_attr_for_new_default_value_test,
```
Comment (skhomuti) [inline tests/functional/test_config_after_upgrade.py:186]: provide ids for params

**Review by skhomuti — CHANGES_REQUESTED** [review]

File: tests/functional/test_config_after_upgrade.py (hunk 1 above)
Comment (dgusakov) [reply tests/functional/test_config_after_upgrade.py:200]: done

**Review by skhomuti — APPROVED** [review]

File: tests/functional/test_config_after_upgrade.py [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+from allure_commons.types import AttachmentType
+from tests.functional.plugin_utils import AnyADCMObject
+
+
+###############################
+# Tools
+###############################
+
+
+def _update_config_and_attr_for_new_field_test(config: OrderedDict, attr: OrderedDict, group_config=False):
```
Comment (alexterent) [inline tests/functional/test_config_after_upgrade.py:31]: Should you add docstrings and hints for return? 
And in other functions in this file.

File: tests/functional/test_config_after_upgrade_data/provider_for_upgrade_with_group_configs_with_changed_group_customisation/config.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      type: map
+      required: false
+      default:
+        age: '24'
+        name: Joe
+        sex: m
+
+- type: host
+  name: host_from_provider
+  version: 1
```
Comment (alexterent) [inline tests/functional/test_config_after_upgrade_data/provider_for_upgrade_with_group_configs_with_changed_group_customisation/config.yaml:109]: add line

File: tests/functional/test_config_after_upgrade.py (hunk 2 above)
Comment (dgusakov) [reply tests/functional/test_config_after_upgrade.py:31]: These are local service functions. I don't think that docstrings will add much here

File: tests/functional/test_config_after_upgrade_data/provider_for_upgrade_with_group_configs_with_changed_group_customisation/config.yaml (hunk 3 above)
Comment (dgusakov) [reply tests/functional/test_config_after_upgrade_data/provider_for_upgrade_with_group_configs_with_changed_group_customisation/config.yaml]: Fixed

File: tests/functional/test_config_after_upgrade.py (hunk 2 above)
Comment (dgusakov) [reply tests/functional/test_config_after_upgrade.py:31]: return hints were added

**Review by alexterent — APPROVED** [review]

**Review by skhomuti — APPROVED** [review]

**Pull request merged**